src/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,25 @@
+import {Image} from 'react-native-compressor';
+import {composeImageMeta} from '../utils/string/file';
+import {uploadPhoto} from '../service/file';
+
+const useUploadImage = () => {
+  const uploadPhotoImage = async (pathImg: string) => {
+    try {
+      const compressionResult = await Image.compress(pathImg, {
+        compressionMethod: 'auto'
+      });
+      const asset = new FormData();
+      asset.append('photo', composeImageMeta(compressionResult));
+      const responseUpload = await uploadPhoto(asset);
+      return responseUpload;
+    } catch (e) {
+      return undefined;
+    }
+  };
+
+  return {
+    uploadPhotoImage
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useUploadImage` function is not a hook, it's just a regular function. Hooks in React are functions that start with "use" and allow you to use React state and other React features without writing a class. This function does not use any of these features. It would be more appropriate to rename this function to something like `createUploadImageFunction`.

```diff
- const useUploadImage = () => {
+ const createUploadImageFunction = () => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->